htdocs/user/card.php

```diff
@@ -548,13 +548,8 @@
 					$ret = $object->update($user);		// This may include call to setPassword if password has changed
 					if ($ret < 0) {
 						$error++;
-						if ($db->errno() == 'DB_ERROR_RECORD_ALREADY_EXISTS') {
-							$langs->load("errors");
-							setEventMessages($langs->trans("ErrorLoginAlreadyExists", $object->login), null, 'errors');
-						} else {
-							setEventMessages($object->error, $object->errors, 'errors');
-							$action = 'edit';
-						}
+						setEventMessages($object->error, $object->errors, 'errors');
```

Comment:
object->error will contains a technical error message. When we have a technical error, we have no choice than showing this. But in the case of an error because of a duplicate value not allowed, we can (so we should) show a message the end user understand.
A solution would be to replace in olde code
setEventMessages($langs->trans("ErrorLoginAlreadyExists", $object->login), null, 'errors');
into
setEventMessages($langs->trans("ErrorUpdateCanceledDueToDuplicatedUniqueValue", $object->login), null, 'errors');   // duplicate value can be a login already exists or a unique extrafield exists.

with ErrorUpdateCanceledDueToDuplicatedUniqueValue=Error, The update was canceled because one of the pieces of data that should be unique already exists